runconfig/opts/parse.go

```diff
@@ -248,6 +249,12 @@ func Parse(cmd *flag.FlagSet, args []string) (*container.Config, *container.Host
 		hostname = parts[0]
 		domainname = parts[1]
 	}
+	if hostname != "" {
+		matched, _ := regexp.MatchString("^(([[:alnum:]]|[[:alnum:]][[:alnum:]\\-]*[[:alnum:]])\\.)*([[:alnum:]]|[[:alnum:]][[:alnum:]\\-]*[[:alnum:]])$", hostname)
```

Comment:
Perhaps we should add a comment here, explaining what RFC we're validating against?, e.g.

    // Validate if the given hostname is RFC 1123 (https://tools.ietf.org/html/rfc1123) compliant.